docs/_docs/30-fasterkv-manual-locking.md

```diff
@@ -0,0 +1,266 @@
+
+As a terminology note, the sub-chain of r#### records is referred to as the `ReadCache` prefix of that hash chain.
+
+In FASTER v1, updates involving `ReadCache` records strip the entire `ReadCache` prefix from the chain. Additionally, the `ReadCache` prefix is stripped from the hash chain when a `ReadCache` page with that hashcode is evicted due to memory limits. In FASTER v2, because `ReadCache` records may be locked, we must not lose those locks. This is resolved in two ways:
+- On record updates, `ReadCache` prefixes are preserved except for the specific record being updated, which is spliced out and transferred to a `CopyToTail` on the main log, including any locks.
+- When `ReadCache` pages are evicted, their records are removed from the `ReadCache` prefix, and any with locks are transferred to the `LockTable`.
+
+### Record Transfers
+
+In normal FASTER operation, records are appended at the tail of the log and do not move. The `HashTable` points to these records for each distinct hash code.
+
+Record transfers occur when a ReadCache entry must be updated, or a record is evicted from either ReadCache or the main log while it holds locks.
+
+#### `ReadCache` Records at Tail of Log
+Note that this resolution is only needed if there is an active `ManualFasterOperations` session at the time `ReadCacheEvict` is called. However, we must already traverse the `ReadCache` records, and it is possible for a new `ManualFasterOperations` session to start during the duration of `ReadCacheEvict`, so we do not optimize for the no-`ManualFasterOperations` case.
+
+For brevity, `ReadCache` is abbreviated RC, `CopyToTail` is abbreviated CTT, and `LockTable` is abbreviated LT. Main refers to the main log. The "final RC record" is the one at the RC->Main log boundary. As always, avoiding locking cost is a primary concern. 
+
+For record transfers involving the ReadCache, we have the following high-level considerations:
+- There is no record-transfer concern if the first record in the chain is not a `ReadCache` entry (normal CAS will do all necessary concurrency control as presently, and there are no outsplices).
+  - Otherwise, we insplice between the final RC entry and the first main-log entry; we never splice into the middle of the RC prefix chain.
+- Even when RC entries are present in the tail, we must avoid latching because it affects all insert operations (at a minimum).
+- "Insplicing" occurs when a new record is inserted into the main log after the end of the ReadCache prefix string.
+- "Outsplicing" occurs when a record is spliced out of the RC portion of the hash chain (main log records are never spliced out) because the value for that key must be updated, or because we are evicting records from the ReadCache. Outsplicing introduces concurrency considerations but we must support it; we cannot simply mark ReadCache entries as Invalid and leave them there, or the chain will grow without bound. 
```

Comment:
More precisely, records are destroyed at read cache eviction. chain will not grow without bound, it will get irretrievably broken.